contracts/samples/SimpleAccount.sol

```diff
@@ -80,9 +79,14 @@ contract SimpleAccount is BaseAccount {
      * an account must have a method for replacing the entryPoint, in case the the entryPoint is
      * upgraded to a newer version.
      */
-    function _updateEntryPoint(address newEntryPoint) internal override {
-        emit EntryPointChanged(address(_entryPoint), newEntryPoint);
-        _entryPoint = IEntryPoint(payable(newEntryPoint));
+    function initialize(IEntryPoint anEntryPoint, address anOwner) public virtual initializer {
+        _initialize(anEntryPoint, anOwner);
+    }
+
+    function _initialize(IEntryPoint anEntryPoint, address anOwner) internal virtual {
+        _entryPoint = anEntryPoint;
```

Comment:
initializer function can only be called once - even when replacing implementation.
This means that your wallet can't change neither its entryPoint nor its owner.
Please add samples on how to use your wallet to change owner and entryPoint

Reply:
I think that:
- entryPoint should be immutable member of the impl.
- the factory should have a method getImpl(entryPoint),  (and a hash of impl by entryPoint)
- this also means factory calls "new Impl" by itself, and doesn't receive it as constructor param.
- anyone can call addImpl() to create a new entry (whoever calls it pays for the gas to add it to the hash. its not "official" or "endorsed" entryPonit)
- setEntryPoint in SimpleAccount can be implemented as
    setImplementation(factory.getImpl(newEntryPoint))
   (this means that implementation has 2 immutable members (and constructor params): factory and entrypoint. the factory is interface, since you can't make circular reference to the class that creates you.